docs/patterns.rst

```diff
@@ -114,6 +118,86 @@ a ``categorical`` slot.
 
 Rasa automatically represents (featurises) this as a one-hot encoding of the values: ``(1,0,0)``, ``(0,1,0)``, or ``(0,0,1)``.
 
+**Slot features**
+
+.. _slot_features:
+
+When Rasa Core runs training against your story, the presence of a ``Slot`` entry will be used to help 
+determine the next action that should be taken. 
+
+This works best with ``CategoricalSlot`` slot types. A ``TextSlot`` can have any value, but it only has one 
+feature - set ``(1)`` or not set ``(0)``. A ``CategoricalSlot`` has a set number of values and a feature for each. Rasa core 
+will be able to make decisions based not only on whether the value is set but also on the value itself.
+
+.. code-block:: yaml
+
+    restaurant_availability:
+        type: categorical
+        values:
+        - unknown
+        - booked-out
+        - waiting-list
+        - available
+
+When the ``restaurant_availability`` slot is set Rasa Core will be able to determine if the restaurant in question is
+available and choose radically different actions to perform based on the value.
+
+The ``Slot`` *might* be set by Rasa Core itself from entities detected by the NLU module, but usually you would 
+return the value of the ``Slot`` from your ``Action`` and then use the next ``Turn`` of the conversation to
+check what feature is set.
+
+.. code-block:: python
+
+    def run(self, dispatcher, tracker, domain):
+        # some logic here to decide if the restaurant is "available", "booked-out" or whatever
+        return [SlotSet("restaurant_availability", "available")]
+
+In this first story we will try and make a booking for 5 people in a restaurant on the night of 21st August 2018. 
+In this case the restaurant is booked out so we want to apologise to the customer and suggests similar restaurants.
```

Comment:
Make a comment here that writing these stories out by hand, and they are shown for illustration. Using interactive learning to generate these stories is *much* less error-prone

Reply:
interesting, that makes sense, but I hadn't realised it made such a difference